arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedDictionaryEncodedParquetValuesReader.java

```diff
@@ -59,14 +59,10 @@ public void nextBatch(
         }
         int numValues = Math.min(left, currentCount);
         for (int i = 0; i < numValues; i++) {
-          int index = idx * typeWidth;
-          if (typeWidth == -1) {
-            index = idx;
-          }
```

Comment:
This happens in the case of var width binary, so the call to `VarWidthBinaryDictEncodedReader::nextVal `works correctly in that case.
Of course, it still works correctly with the change.